src/BuildingRegistry.Projections.Extract/BuildingUnitAddressLinkExtract/BuildingUnitAddressLinkExtractProjections.cs

```diff
@@ -84,6 +84,26 @@ await context
             {
                 await RemoveLink(context, message.Message.BuildingUnitPersistentLocalId, message.Message.AddressPersistentLocalId, ct);
             });
+
+            When<Envelope<BuildingUnitAddressWasReplacedBecauseAddressWasReaddressed>>(async (context, message, ct) =>
+            {
+                await RemoveLink(context, message.Message.BuildingUnitPersistentLocalId, message.Message.SourceAddressPersistentLocalId, ct);
```

Comment:
In the domain, we check if the relation to remove and the relation to add exist. I would do the same here.